forum/views/comments.py

```diff
@@ -0,0 +1,209 @@
+
+from typing import Any, Optional
+
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import status
+from rest_framework.permissions import AllowAny
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.serializers import ValidationError
+from rest_framework.views import APIView
+
+from forum.models import Comment
+from forum.serializers.comments_api import CommentsAPISerializer
+from forum.utils import str_to_bool
+
+
+class CommentsAPIView(APIView):
+    """
+    API View to handle GET, POST, PUT, and DELETE requests for comments.
+    """
+
+    permission_classes = (AllowAny,)
+
+    def _validate_comment(self, comment_id: str) -> dict[str, Any]:
+        """
+        Validates the comment if it exists or not.
+
+        Parameters:
+            comment_id: The ID of the comment.
+        Response:
+            comment object for the given comment_id.
+        """
+        comment = Comment().get(comment_id)
+        if not comment:
+            raise ObjectDoesNotExist
+        return comment
+
+    def get(self, request: Request, comment_id: str) -> Response:
+        """
+        Retrieves a parent comment.
+        For chile comments, below API is called that return all child comments in children field
+        url: http://localhost:8000/forum/api/v2/threads/66ab94950dead7001deb947a
+
+        Parameters:
+            request (Request): The incoming request.
+            comment_id: The ID of the comment.
+        Body:
+            Empty.
+        Response:
+            The details of the comment for the given comment_id.
+        """
+        try:
+            comment = self._validate_comment(comment_id)
+        except ObjectDoesNotExist:
+            return Response(
+                {"error": "Comment does not exist"}, status=status.HTTP_404_NOT_FOUND
+            )
+        data = self._prepare_response(comment, exclude_fields=["sk"])
+        return Response(data, status=status.HTTP_200_OK)
+
+    def post(self, request: Request, comment_id: str) -> Response:
+        """
+        Creates a new child comment.
+        For parent comment below API is called.
+        url: http://localhost:8000/forum/api/v2/threads/66ab94950dead7001deb947a/comments
+
+        Parameters:
+            request (Request): The incoming request.
+            comment_id: The ID of the parent comment for creating it's child comment.
+        Body:
+            body: The content of the comment.
+            course_id: The Id of the respective course.
+            user_id: The requesting user id.
+            anonymous: anonymous flag(True or False).
+            anonymous_to_peers: anonymous to peers flag(True or False).
+        Response:
+            The details of the comment that is created.
+        """
+        data = request.data
+        new_comment_id = Comment().insert(
+            body=data["body"],
+            course_id=data["course_id"],
```

Comment:
maybe we can validate request data before accessing it to avoid code breaks.
Alternatively, if these are optional, we should use `dict.get` here too.

Reply:
actually these are not optional that's y I kept it this way to break if it doesn't exists. `I can add some validations`.

Most of the data is filled in our model's methods, it won't make sense to serialize it first and then again serializing it to get the desired result as some fields come from serializer.